pages/api/user/[id]/index.ts

```diff
@@ -10,20 +11,22 @@ export default async function handler(
   req: NextApiRequest,
   res: NextApiResponse
 ) {
-  const {user, isGlobalAdmin} = await getCurrentUser(req, res)
+  const {user: currentUser, isGlobalAdmin} = await getCurrentUser(req, res)
 
   switch (req.method) {
     case 'GET':
       try {
+        const user = await User.findById(req.query.id)
+
         return res
-          .status(400)
-          .json({ success: false, message: 'not implemented' })
+          .status(200)
+          .json({ success: true, data: user })
       } catch (error) {
         return res.status(400).json({ success: false })
       }
     case 'PATCH':
       try {
-        if (req.query.id !== user?._id.toString() && !isGlobalAdmin) {
+        if (req.query.id !== currentUser?._id.toString() && !isGlobalAdmin) {
```

Comment:
`?.` makes no sense in `currentUser?._id.toString()` because of `.toString()` invocation. replace with `currentUser?._id?.toString()`